.github/workflows/main.yml

```diff
@@ -65,7 +65,7 @@ jobs:
     timeout-minutes: 60
     strategy:
       matrix:
-        python-version: ['3.8', '3.10', '3.11']
+        python-version: ['3.8', '3.10', '3.11', '3.12']
```

Comment:
Per @speth's comment, please remove `'3.12'`.

Reply:
Why not duplicate this job into `post-merge` with only 3.12? Then we could avoid changing Python version and OS in the same job, to better isolate build failures.

Reply:
You could add both in the post-merge tests. I'm not even sure which would be more likely to succeed. The logs in #1602 seem to indicate that the only reason that this doesn't work on Python 3.12 in that case is because Cantera is explicitly ignoring it.